newdle/client/src/UserMenu.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import {useSelector} from 'react-redux';
+import {Dropdown, Icon} from 'semantic-ui-react';
+import Gravatar from 'react-gravatar';
+import {useAuthentication} from './auth';
+import {isLoggedIn, getUserInfo} from './selectors';
+
+import styles from './UserMenu.module.scss';
+
+const defaultUIAvatarsArgs = [
+  40, // size
+  'f0e9e9', // background
+  '8b5d5d', // color
+  2, // length
+  0.5, // font size
+  true, // rounded
+];
+
+const NoUserAvatar = () => (
+  <Icon bordered inverted color="grey" name="user" size="large" circular />
+);
+
+function renderGravatar({email, first_name: firstName, last_name: lastName, full_name: fullName}) {
+  const gravatarFallback =
+    firstName && lastName
+      ? encodeURI(
+          `//ui-avatars.com/api/${firstName[0]} ${lastName[0]}/${defaultUIAvatarsArgs.join('/')}`
+        )
+      : 'mp';
+
+  return (
+    email && (
```

Comment:
`if (!email) return null;` at the beginning.. the fallback defined above isn't even used then.

However, I don't think we actually need it: If we have a user we always have an email as well.